src/main/java/org/reactome/release/goupdate/GoUpdateInstanceEditUtils.java

```diff
@@ -0,0 +1,103 @@
+package org.reactome.release.goupdate;
+
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.gk.model.GKInstance;
+import org.gk.persistence.MySQLAdaptor;
+import org.reactome.release.common.database.InstanceEditUtils;
+
+/**
+ * Some utility methods for working with InstanceEdits for GO Update.
+ * @author sshorser
+ *
+ */
+class GoUpdateInstanceEditUtils
+{
+	/**
+	 * Different types of instance edits.
+	 * @author sshorser
+	 *
+	 */
+	enum GOUpdateInstEditType
+	{
+		NEW("New GO term was created"),
+		MODIFIED("GO term attributes were modified"),
+		REF_CLEARED("Attribute referring to a GO term has been cleared"),
+		REF_ATTRIB_UPDATE("Attribute referring to a GO term has been set to a *different* GO term"),
+		DISPLAY_NAME("Display Name updated because a GO term was updated"),
+		UPDATE_RELATIONSHIP("GO Term relationships were updated");
```

Comment:
I'm curious about how you came up with these different IE types -- were the types apparent from the code or did you have to change much of the code to have the IE types conform to a specific type?

Reply:
Code structure did not need to change much. I looked at the code for the different places where InstanceEdits were used and made an InstanceEditType for the different scenarios.